src/client.rs

```diff
@@ -142,10 +143,48 @@ pub struct CachedFeature {
     // on submission will be the next logical progression.
     enabled: AtomicU64,
     disabled: AtomicU64,
+    variant_metrics: DashMap<String, AtomicU64>,
     // Variants for use with get_variant
     variants: Vec<api::Variant>,
 }
 
+impl CachedFeature {
+    fn count_variant(&self, variant_name: &str) {
+        self.variant_metrics
+            .entry(variant_name.into())
+            .and_modify(|count| {
+                count.fetch_add(1, Ordering::Relaxed);
+            })
+            .or_insert(AtomicU64::new(1));
+    }
+}
+
+impl From<&CachedFeature> for ToggleMetrics {
+    fn from(feature: &CachedFeature) -> Self {
+        fn convert_variants_count(original: &DashMap<String, AtomicU64>) -> HashMap<String, u64> {
```

Comment:
I want to say this could be more idiomatically expressed as:

```fn convert_variants_count(original: &DashMap<String, AtomicU64>) -> HashMap<String, u64> {
    original
        .iter()
        .map(|item| {
            let (key, value) = item.pair();
            let count = value.load(Ordering::Relaxed);
            (key.clone(), count)
        })
        .collect()
}```

Not tested, just eyeballed. And this may be a matter of taste so up to you

Reply:
Nah, I like that a lot! Super catch 🙌🏼